src/CommunityToolkit.Maui/Views/RatingView/RatingView.shared.cs

```diff
@@ -0,0 +1,627 @@
+	/// <param name="ratingItems">A list of rating item visual elements to update.</param>
+	/// <param name="rating">Current rating value (e.g., 3.7)</param>
+	/// <param name="filledColor">Color of a filled item.</param>
+	/// <param name="emptyColor">Color of an empty item.</param>
+	/// <param name="backgroundColor">Color of the item.</param>
+	static void UpdateRatingItemColors(List<VisualElement> ratingItems, double rating, Color filledColor, Color emptyColor, Color backgroundColor)
+	{
+		int fullShapes = (int)Math.Floor(rating); // Determine the number of fully filled shapes
+		double partialFill = rating - fullShapes; // Determine the fraction for the partially filled shape (if any)
+		backgroundColor ??= Colors.Transparent;
+		for (int i = 0; i < ratingItems.Count; i++)
+		{
+			((Shape)((Border)ratingItems[i]).Content!).Fill = emptyColor;
+			if (i < fullShapes)
+			{
+				ratingItems[i].BackgroundColor = filledColor; // Fully filled shape
+				ratingItems[i].Background = null;  // Empty fill
+			}
+			else if (i == fullShapes && partialFill > 0)
+			{
+				ratingItems[i].Background = new LinearGradientBrush(
+					[
+							new GradientStop(filledColor, 0),
+							new GradientStop(filledColor, (float)partialFill),
+							new GradientStop(backgroundColor, (float)partialFill),
+					],
+					new Point(0, 0), new Point(1, 0));  // Partial fill
+			}
+			else
+			{
+				ratingItems[i].BackgroundColor = backgroundColor;  // Empty fill
+				ratingItems[i].Background = null;  // Empty fill
+			}
+		}
+	}
+
+	/// <summary>Updates each rating item shape colors based on the rating value (filled, partially filled, empty).</summary>
+	/// <remarks>At this time, since .NET MAUI doesn't have direct partial fill, we approximate with a gradient.</remarks>
+	/// <param name="ratingItems">A list of rating item shape visual elements to update.</param>
+	/// <param name="rating">Current rating value (e.g., 3.7)</param>
+	/// <param name="filledColor">Color of a filled shape.</param>
+	/// <param name="emptyColor">Color of an empty shape.</param>
+	static void UpdateRatingShapeColors(List<VisualElement> ratingItems, double rating, Color filledColor, Color emptyColor)
+	{
+		int fullShapes = (int)Math.Floor(rating); // Determine the number of fully filled shapes
+		double partialFill = rating - fullShapes; // Determine the fraction for the partially filled shape (if any)
+		for (int i = 0; i < ratingItems.Count; i++)
+		{
+			Shape ratingShape = (Shape)ratingItems[i];
+			if (i < fullShapes)
+			{
+				ratingShape.Fill = filledColor; // Fully filled shape
+			}
+			else if (i == fullShapes && partialFill > 0)
+			{
+				ratingShape.Fill = new LinearGradientBrush(
+					[
+							new GradientStop(filledColor, 0),
+							new GradientStop(filledColor, (float)partialFill),
+							new GradientStop(emptyColor, (float)partialFill),
+					],
+					new Point(0, 0), new Point(1, 0)); // Partial fill
```

Comment:
this `LinearGradientBrush` construct, is almost the same as the one in at line 470, can we refactor it be a method? That way will be one code to maintain.

Reply:
Refactored out to a method.